Feature/organize with ecosystem template

Unathi-Skosana opened the pull request:
### Change log

- Reorganized repository to resemble [qiskit-ecosystem_template](https://github.com/mickahell/qiskit-ecosystem_template) with a few minor differences, _i.e._ `src` folder was renamed to `qpong`.
- Renamed class method in `circuit_grid_model.py` to `compute_circuit` to `construct_circuit`, i think the latter is a standard name
- Simplified the aforementioned class method.

HuangJunye commented on qpong/containers/vbox.py:
```diff
@@ -18,17 +20,25 @@
+    Arranges sprites vertically
+    """
+
+    def __init__(self, xpos, ypos, orientation, *sprites):
```
Are you trying to use orientation so that we can just use one VBOX to replace both VBOX and HBOX?

HuangJunye replied:
Removed the changes and revert back to original.

Unathi-Skosana replied:
Many thanks for reviewing the PR, and subsequent modifications . I will try to closely follow your advise in subsequent PRs for a clear separation of changes and logic.

The changes to VBox and HBox were not meant to make it in the PR (sorry about that). I also noticed that HBox isn't used at all, hence to appease pylint, I meant revert back VBox to its original state and remove HBox entirely.

HuangJunye reviewed (commented):
Thanks a lot for working on this. I made some changes and managed to make the game worked again. The main.py is now outside of the qpong folder. That's the way I got it working again. I think it may work if you try to move it back inside `qpong` folder and use relative imports in all files.

I also reverted the changes of VBOX and HBOX because it's causing some troubles. Because you added a new keyword arguments and didn't modify the code where VBOX is used. I think the change is not critical because HBOX is never used. You can just keep it as is.

In general, I would advise not to mix different changes in the same PR. It would make review much more difficult. This PR should only introduce changes in order to add the template.

I see `circuit_grid`, `circuit_grid_model` have been modified extensively. I will need to take a look at them another day to make sure everything is OK. 

And this time I haven't reviewed anything related to the ecosystem template. I will take a look sometime next week. 

Meanwhile can you make changes to `.pylintrc` to relax the linting requirements and make the tests pass? I think you may also need to add docstrings for the HBOX that was added back.

Unathi-Skosana commented:
- Added doc strings
- Relaxed some of pylint's restrictions (_i.e_ too-many-statements, too-many-branches _etc pp._) 
- Subsequently discovered and fixed out of bounds errors related to circuit grid

HuangJunye commented:
This PR is superseded by #57
